src/profile-logic/profile-data.js

```diff
@@ -685,6 +688,21 @@ export function toValidImplementationFilter(
   }
 }
 
+export function toValidCallTreeSummaryStrategy(
+  strategy: mixed
+): CallTreeSummaryStrategy {
+  switch (strategy) {
+    case 'timing':
+    case 'js-allocations':
+    case 'native-allocations':
+      return strategy;
+    default:
+      // Default to timing if the strategy is not recognized. This value can come
+      // from a user-generated URL.
```

Comment:
What is user-generated URL? Inside `urlStateToUrlObject` we are doing
```js
query.ctSummary =
  urlState.profileSpecific.callTreeSummaryStrategy === 'timing'
    ? undefined
    : urlState.profileSpecific.callTreeSummaryStrategy;
```
So `timing` is not going to be serialized in any case, right?

Reply:
That ternary means basically: if the strategy is "timing", do not URL serialize it, since it's the default. If it's any of the other `CallTreeSummaryStrategy` values, then serialize it to the URL, as it's non-default. This way we can keep the URL shorter for default views, which are the common case.